src/startupscriptgenerator/src/dotnetcore/scriptgenerator.go

```diff
@@ -42,6 +42,16 @@ func (gen *DotnetCoreStartupScriptGenerator) GenerateEntrypointScript(scriptBuil
 	}
 
 	dotnetBinary := "/usr/share/dotnet/dotnet"
+
+	preRunScriptPath := common.GetPathEnvironmentVariable(consts.AppServicePreRunScriptFilePathEnvVarName)
+	if preRunScriptPath != "" && !common.PathExists(dotnetBinary) {
+		scriptBuilder.WriteString("echo 'Running the user provided pre-run command within pre-run script... ")
+		preRunScriptFileDir := filepath.Dir(preRunScriptPath)
+		scriptBuilder.WriteString("cd \"" + preRunScriptFileDir + "\"\n")
+		preRunCommand := "bash \"" + preRunScriptPath + "\""
```

Comment:
A user could specify shebang in their script file, so we do not need to explicit mention this?